### DISCUSSION TITLE
Error in Split Brain Mask 1 "RuntimeError: System command exited with non null value : 111"

### DISCUSSION THREAD

**Discussion by ryode-ut** [brainvisa/brainvisa.github.io#184] [Q&A] [answered]
Hello everyone,

I'm a beginner with the BrainVisa tool and I'm trying to use the Morphologist 2021 and BrainVisa 5.1.1.

I imported T1W MRI data of a subject and ran the Morphologist 2021, then the error message appeared. It seems that the skull stripping process didn't run correctly because \t1mri\default_acquisition\default_analysis\segmentation\skull_stripped_subject_T1.nii.gz contains the skull component.

I suspect that the subject has an artificial tooth and that causes the artifact that led to the error in the process.

Can I do anything to make it run correctly?
I hope you can give me some advice. I'd appreciate it.

Thank you very much.
Ryoki.

**Comment by denisri** [comment]
Hi,
It's difficult to answer without any view of the problem. If you can provide a snapshot of the raw MRI and/or the first step which goes wrong, it could help.
Generally when the segmentation goes wrong, the weakest step is the bias correction, which can have consequences on later steps. You may check that step (looking at the doc to see what is expected and could go wrong), then check the brain mask. Skull stripping consists in masking the image with the brain mask, so I suspect this step did not work. Moreover you have an error in the split brain mask step, which consists in separating this brain mask into 3 parts. If it fails it may be because the mask is wrong.
Denis

**Comment by ryode-ut** [comment]
Hi Denis,

I appreciate your comment. I took some pictures of my process and the MRI data. The MRI images are the skull-stripped one and the original T1W data. I hope they help you to understand the situation. I'd like to mention that the subject was a post-stroke patient.
I agree with your comment that something's going wrong in the split brain step because of the incorrect brain mask. Do you have any idea how to deal with this?

Thank you very much.
Ryoki

![image](https://github.com/user-attachments/assets/40e96002-aa33-4ce2-9304-a4ea83df26f1)
![image (1)](https://github.com/user-attachments/assets/7a25fcfa-de33-455e-97ca-78371dd88148)
![image (2)](https://github.com/user-attachments/assets/08bc6d2b-98fc-4d3b-8512-fe597421f1fb)
![image (3)](https://github.com/user-attachments/assets/3ca7ec4e-61e2-459e-8abb-c9f699f9f0d5)

**Comment by denisri** [comment]
Hi,
The subject has a stroke, it might be the cause of the problem, but I still cannot help much. The image globally doesn't look so bad, but there is actually a black zone. I would need to see the visualization of the "t1mri_nobias", and in the "compute brain mask" step, the "brain_mask" images (click on the eye icons for these parameters to have the visualization).
Note also that if we cannot make Morphologist work for this image, and if Freesurfer can, then we also have the possibility to import Freesurfer segmentation into BrainVisa/Morphologist, and process the sulci further.
Denis

**Comment by ryode-ut** [comment]
Hi Denis

Thanks again for your comment. I used the BrainVisa viewer this time and I hope it meets your expectations.
Could you tell me what the "compute brain mask" step means in more detail? You'd like me to show you the LINUX command lines during the computing brain mask process?

For the same subject data, the Freesurfer could work well. I have also attached the image of the Freesurfer result.

I'd really appreciate your interest.

Ryoki

![image](https://github.com/user-attachments/assets/d3de7982-00cb-4ba5-ac9d-1924c0d20d74)

![image](https://github.com/user-attachments/assets/30d54f43-5d1b-46dd-b9ce-7da47e2c3522)

**Comment by denisri** [comment]
"Compute brain mask" should extract the brain (whole brain, with grey and white matters, and cerebellum). Here we see that it has failed and it has "leaked" in the skull, scalp and the inferior parts under the brain.
Could you please also show the "t1mri_nobias" visualization, because I suspect this earlier step has not worked well also. (bias correction is also called "non-uniformity correction" or "NUC" or "inhomogeneity correction" in some software).
The "T1 Bias Correction" step of the Morphologist pipeline has a few parameters that can be changed and which sometimes help: you can try changing the "sampling" and 'field rigidity" parameters (I generally try smaller values then the bias field correction is not correct, using half of the default for instance).

Freesurfer has succeeded, indeed (except around the stroke, where grey and pial meshes have shift outwards).
If we cannot process this brain using Morphologist, then you can use the BrainVisa process in the "Freesufer" toolbox of BrainVisa: "Freesurfer/Import/Import Freesurfer grey/white segmentation to Morphologist". As it says, it will import Freesurfer segmentation, then run the remaining steps of Morphologist.

Denis

**Comment by ryode-ut** [comment]
Hi Denis

First of all, sorry for the late reply recently.

The figure below shows the bias correction view. Does it look done correctly?

As you mentioned, some meshes don't look good in the Freesurfer.
I'll try to proceed the BrainVISA process using the Freesurfer data.

Thanks for your concern.

Ryoki

![image](https://github.com/user-attachments/assets/14e24e7e-d5ba-4a78-8ed0-9798fb53bbd9)

**Comment by denisri** [comment]
Outside of the stroke, it doesn't look so bad. The initial image doesn't seem to have a strong bias either. So maybe the stroke induced a "too strong" correction, which results in bringing intensities of grey and white matters too close. There are 2 things I would try if I were you:
- skip the bias correction: copy the raw (initial) image to the location of the bias corrected one (the "nobias" image), then un-select the bias correction and previous steps in the pipeline, and run it again
- or try increasing the field rigidity parameter in the bias correction step, try doubling it, for instance. You can un-select the steps before bias correction (orientation/normalization) to save time, but keep the bias correction selected.
Anyway even if it works, seeing the altered intensities around the stroke, I guess Morphologist will not do better than Freesurfer at the end in this region, so if you can use the Freesurfer segmentation, it will be OK also.

**Comment by ryode-ut** [comment]
Hi Denis

Thank you for comments again. 

I forgot to tell you the important stuff. I used the initial image which already got the bias correction by the CHARM program in the SimNIBS software. So, this image got the bias correction twice.
You'd guess that several bias correction processes might make it difficult to do the skull-stripped process, am I right?

My analysis process so far conducted in this way;
first, conducting the CHARM program using raw data;
second, conducting the Freesurfer program using the bias correction data created by the CHARM;
third, conducting the Morphologist using the same bias correction data.

I didn't know that there is Freesurfer/Morphologist pipeline and Freesurfer may be better than Morphologist in terms of the process capability. I'll try to do your suggestions.

If you don't mind, could you tell me why you think Freesurfer is better than Morphologist?
It'll be really helpful to expand my knowledge. I've been doing these stuff alone.

Thank you.

Ryoki

**Comment by denisri** [comment]
Well, generally it doesn't harm to do the bias correction twice, at least using the same algorithm: as the algo searches for an optimum correction, running it again will likely stick with the same optimum. But 1. when it's not the same algorithm, the optimality criteria may differ; 2. when there is an "anomaly" on the image which may disturb the algorithm's notion of gray and white matters intensity definition, the intensities to be bias-corrected may shift. In both cases the algo may over-correct and erase (or reduce), not only the low frequency shifts in the image for a given (grey/white/) matter, but also reduce the contrast between tissues, or blur it. Here we are in both situations.
Otherwise I am not saying that Freesurfer performs generally better than Morphologist, both work quite well normally. I was just saying that given the stroke and hyper-intensities around it in this particular image, which make grey matter look like white matter (I think), I suspect that Morphologist will not segment properly GM and WM around this region also, thus will at best do more or less the same as what Freesurfer has done, here.

**Comment by ryode-ut** [comment]
Hi Denis

I'd appreciate your concern. Now I understand what you meant about the Freesurfer and Morphologist comparison.
So far, as I mentioned, I've repeated the bias correction process using the several programs. I'll reconsider the methodology, and try to do your suggestions to make segmentation process better with the Morphologist.

When it comes to doing your suggestions, is there any manual or instruction to change the analyzing settings of the Morphologist, like skipping the bias correction step?
If you don't mind, I'd like you to give me any comment on that. I searched but couldn't find it. 

Thank you.

Ryoki

**Comment by denisri** [comment, marked as answer]
When you open the Morphologist pipeline in Brainvisa interface, in the left panel you have the different steps of the pipeline (the sub-processes making the pipeline). Each of them has a check button so that you can select or un-select them. When running, the un-selected sub-processes are not included in the workflow.
Denis

**Comment by ryode-ut** [comment]
Hi Denis.
I'd really appreciate all of your thoughtful comments. I'll do my best in my research.
I extend my apologies for the late response.
Ryoki
